File: Examples/DotNET/CSharp/WatermarksOrStamps/AddPageNumberStamp.cs
using System;
using Com.Aspose.PDF.Api;
using Com.Aspose.PDF.Model;
using Com.Aspose.Storage.Api;

namespace WatermarksOrStamps
{
    class AddPageNumberStamp
    {
        public static void Run()
        {
            // ExStart:1
            PdfApi pdfApi = new PdfApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "sample-input.pdf";
            int pageNumber = 1;
            String storage = "";
            String folder = "";

            Stamp body = new Stamp();
            body.Background = true;
            body.Value = "Page # of 2";
            body.Type = "PageNumber";
            body.PageIndex = 0;
            body.LeftMargin = 0.0;
            body.Opacity = 0.5;
            body.RightMargin = 0.0;
            body.TopMargin = 0.0;
            body.YIndent = 100.0;
            body.XIndent = 100.0;
            body.Zoom = 1.0;
            body.TextState = null;
            body.Width = 300.0;
            body.Height = 300.0;
            body.StartingNumber = 1;

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));

                // Invoke Aspose.PDF Cloud SDK API to add page number stamp to a pdf page
                SaaSposeResponse apiResponse = pdfApi.PutPageAddStamp(fileName, pageNumber, storage, folder, body);

                if (apiResponse != null)
                {
                    Console.WriteLine("Add Page Number Stamp to a PDF Page, Done!");
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);

            }
            // ExEnd:1
        }
    }
}

File: Examples/Perl/WatermarksOrStamps/AddPageNumberStamp.pl
#ExStart:1
use lib 'lib';
use strict;
use warnings;
use utf8;
use File::Slurp; # From CPAN
use JSON;

use AsposeStorageCloud::StorageApi;
use AsposeStorageCloud::ApiClient;
use AsposeStorageCloud::Configuration;

use AsposePdfCloud::PdfApi;
use AsposePdfCloud::ApiClient;
use AsposePdfCloud::Configuration;
use AsposePdfCloud::Object::Stamp;

my $configFile = '../config/config.json';
my $configPropsText = read_file($configFile);
my $configProps = decode_json($configPropsText);
my $data_path = '../../../Data/';
my $out_path = $configProps->{'out_folder'};

$AsposePdfCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposePdfCloud::Configuration::api_key = $configProps->{'api_key'};
$AsposePdfCloud::Configuration::debug = 1;

$AsposeStorageCloud::Configuration::app_sid = $configProps->{'app_sid'};
$AsposeStorageCloud::Configuration::api_key = $configProps->{'api_key'};

# Instantiate Aspose.Storage and Aspose.Pdf API SDK
my $storageApi = AsposeStorageCloud::StorageApi->new();
my $pdfApi = AsposePdfCloud::PdfApi->new();

# Set input file name
my $name =  'SampleImage.pdf';
my $pageNumber =  1;
my @stampBody = AsposePdfCloud::Object::Stamp->new( 
		'PageIndex' => 1,
		'Value' => 'Page # of 2',
		'Type' => 'PageNumber',
		'StartingNumber' =>  1,
		'LeftMargin' =>  0.0,
		'Opacity' =>  0.5,
		'RightMargin' =>  0.0,
		'TopMargin' =>  0.0,
		'YIndent' =>  100.0,
		'XIndent' =>  100.0,
		'Zoom' =>  1.0,
		'Width' =>  300.0,
		'Height' =>  300.0	);

# Upload file to aspose cloud storage 
my $response = $storageApi->PutCreate(Path => $name, file => $data_path.$name);

# Invoke Aspose.Pdf Cloud SDK API to insert page number stamp to a PDF page                             
$response = $pdfApi->PutPageAddStamp(name=>$name, pageNumber=>$pageNumber, body=>@stampBody);

if($response->{'Status'} eq 'OK'){
    # Download document from storage server
    my $output_file = $out_path. $name;
    $response = $storageApi->GetDownload(Path => $name);
	write_file($output_file, { binmode => ":raw" }, $response->{'Content'});
}
#ExEnd:1